src/app/core/services/label.service.ts

```diff
@@ -70,36 +94,94 @@ export class LabelService {
   }
 
   /**
-   * Stores the json data from Github api into the list of arrays in this service
-   * @param labels: the json data of the label
+   * Returns an array of Preset Labels.
    */
-  private populateLabelLists(labels: Array<{}>): void {
+  private getRequiredLabels(): Label[] {
+    const requiredLabels: Label[] = [];
 
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
+    for (const category of Object.keys(this.REQUIRED_LABELS)) {
+      for (const labels of Object.values(this.REQUIRED_LABELS[category])) {
+        requiredLabels.push(labels as Label);
       }
+    }
 
-      this.labelColorMap.set(labelValue, labelColor);
-      const labelList = this.getLabelList(labelType);
+    return requiredLabels;
+  }
 
-      if (labelList !== undefined) {
-        labelList.push({labelValue: labelValue, labelColor: labelColor});
+  /**
+   * Ensures that the repo has the expected labels.
+   * Compares the actual labels in the repo with expected labels. If an expected label is missing,
+   * it is added to the repo. If the expected label exists but the label color is not as expected,
+   * the color is updated. Does not delete actual labels that do not match expected labels.
+   * i.e., the repo might have more labels than the expected labels after this operation.
+   * @param labels: JSON data representing labels in the repo.
+   * @param expectedLabels: expected labels.
+   */
+  private ensureRepoHasExpectedLabels(labels: Array<{}>, expectedLabels: Label[]): void {
+
+    // Parse Input Data to Label[]
+    const labelData: Label[] = this.parseLabelData(labels);
```

Comment:
The function signature is cleaner if both parameters are of type `Label[]`. i.e., this conversion can be done outside of the function.